components/net/tests/http_loader.rs

```diff
@@ -1457,3 +1458,19 @@ fn test_fetch_compressed_response_update_count() {
         (DATA_DECOMPRESSED_LEN + DECODER_BUFFER_SIZE - 1) / DECODER_BUFFER_SIZE;
     assert_eq!(response_update_count, EXPECTED_UPDATE_COUNT);
 }
+
+#[test]
+fn test_origin_serialization_compatability() {
+    let ensure_serialiations_match = |url_string| {
+        let url = Url::parse(url_string).unwrap();
+        let origin = ImmutableOrigin::new(url.origin());
+        let serialized = format!("{}", serialize_origin(&origin));
+        assert_eq!(serialized, origin.ascii_serialization());
+    };
+
+    ensure_serialiations_match("https://example.com");
+    ensure_serialiations_match("http://example.com");
+    ensure_serialiations_match("https://example.com:1234");
+    ensure_serialiations_match("http://example.com:1234");
```

Comment:
Should there be a case for the default port? i.e `ensure_serialiations_match("http://example.com:80");`

Reply:
Good point. I've added this case.